src/Network/NWWebSocketOptions.cs

```diff
@@ -0,0 +1,122 @@
+//
+// NWWebSocketOptions.cs: Bindings the Network nw_browser_t API.
+//
+// Authors:
+//   Manuel de la Pena (mandel@microsoft.com)
+//
+// Copyrigh 2019 Microsoft Inc
+//
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+using OS_nw_protocol_options=System.IntPtr;
+using nw_ws_request_t=System.IntPtr;
+
+namespace Network {
+
+	// this maps to `nw_ws_version_t` in Network.framework/Headers/ws_options.h (and not the enum from NetworkExtension)
+	[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+	public enum NWWebSocketVersion {
+		Invalid = 0,
+		Version13 = 1,
+	}
+
+	[TV (12,0), Mac (10,14), iOS (12,0)]
+	[Watch (6,0)]
+	public class NWWebSocketOptions : NWProtocolOptions {
```

Comment:
@migueldeicaza same here, I think I did choose the correct base class but I'd like some extra pair of eyes.